src/client/Client.cc

```diff
@@ -9192,6 +9192,7 @@ int Client::read(int fd, char *buf, loff_t size, loff_t offset)
   int r = _read(f, offset, size, &bl);
   ldout(cct, 3) << "read(" << fd << ", " << (void*)buf << ", " << size << ", " << offset << ") = " << r << dendl;
   if (r >= 0) {
+    client_lock.unlock();
```

Comment:
This actually is unsafe as the `lock_guard` above will unlock the mutex again when this method exits.

You can instead convert the `lock_guard` above to a scoped lock and then change this to `lock.unlock()`.

Reply:
I have fixed it ,thanks!